### PR TITLE
Merge `Store` and `FeedbackLoop` API. Rename it to `Loop`.

### PR DESCRIPTION
**Pull request by andersio**
This PR merges `Store` and `FeedbackLoop` into one API.

### API renames
* `FeedbackLoop` and `Store` is renamed to `Loop`.
* `Store.view(value:event:)` is renamed to `Loop.scoped(to:event:)`.
* `FeedbackLoop.init(initial:reduce:feedbacks:)` is renamed to `Loop.init(initial:reducer:feedbacks:)`.

### Removed API
* `FeedbackLoop.start()` and `FeedbackLoop.stop()` are now unavailable.
* `Store.state` is now unavailable. It is replaced by `Loop.context`, which is a producer yielding `Context<S, E>`.

### Implementation
`Loop` does type erasure by inheritance internally, so that it can be the currency type for both root loops (global stores) & their scoped views (local derived stores).

This also hides stuff like `start()` and `stop()` from the public API, that were intended for `SignalProducer.feedbackLoop`.

### CODE REVIEW

File: Loop/Public/Loop.swift [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    private let box: LoopBoxBase<State, Event>
+
+    private init(box: LoopBoxBase<State, Event>) {
+        self.box = box
+    }
+
+    public init(
+        initial: State,
+        reducer: @escaping (inout State, Event) -> Void,
+        feedbacks: [Loop<State, Event>.Feedback]
```
Comment (sergdort) [inline Loop/Public/Loop.swift:31]: Should we just remove old feedback and have `Feedback<State, Event>`, I find it quite annoying to have API 

`static func whenSomething() -> Loop<State, Event>. Feedback` vs 
`static func whenSomething() ->  Feedback<State, Event>`

¯\_(ツ)_/¯

Comment (andersio) [reply Loop/Public/Loop.swift:31]: I'd prefer to leave the gradual migration option on the table.

Comment (andersio) [reply Loop/Public/Loop.swift:31]: Alternatively, we can rename the old feedback to something else, so manual migration is required.

Comment (andersio) [reply Loop/Public/Loop.swift:31]: @sergdort lets do that in a separate PR, no?

**Review by sergdort — APPROVED** [review]

**Comment by sergdort** [comment]
Screenshot 2020-05-13 at 21 35 46

**Comment by andersio** [comment]
@sergdort misconfiguration 🤦

**Pull request merged**